comp/collector/collector/collectorimpl/agent_check_metadata.go

```diff
@@ -55,12 +55,18 @@ func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
 
 // GetPayload builds a payload of all the agentchecks metadata
 func (c *collectorImpl) GetPayload(ctx context.Context) *Payload {
-	hostname, _ := hostname.Get(ctx)
+	hostnameData, _ := hostname.Get(ctx)
 
 	meta := hostMetadataUtils.GetMetaFromCache(ctx, c.config)
-	meta.Hostname = hostname
+	meta.Hostname = hostnameData
 
-	cp := hostMetadataUtils.GetCommonPayload(hostname, c.config)
+	legacyResolutionHostnameData, _ := hostname.GetWithProviderWithoutIMDSV2(ctx)
+	if legacyResolutionHostnameData.Hostname != meta.Hostname {
+		meta.LegacyResolutionHostname = legacyResolutionHostnameData.Hostname
+		meta.HostMetaVersion = "1.0"
+	}
+
+	cp := hostMetadataUtils.GetCommonPayload(hostnameData, c.config)
```

Comment:
Since this logic is shared between this and host payload, both of which include CommonPayload, would it make sense to report this data as part of the common payload? This way this IMDSv2 logic would need to be implemented only once.

Reply:
done